### PR TITLE
Get Weather of User's Location

### PR DESCRIPTION
**Pull request by samarthya-jha**
## Why this PR is important 

This PR changes the website so that it displays the weather of the user's location when they first visit the site. It fetches the latitude and longitude of the user, then the city name using [reverse geocoding](https://openweathermap.org/api/geocoding-api) and then calls the OpenWeather API.

## What this PR adds

1. This PR modifies the `App.js` file - an additional Hook is created which executes on page load and fetches the geolocation from the `geolocation.js` file.
2. This PR adds a `geolocation.js` file - using this [Geolocation Package](https://www.npmjs.com/package/@react-native-community/geolocation), `latitude` and `longitude` is fetched using which `cityName` is fetched from [reverse geocoding API](https://openweathermap.org/api/geocoding-api).

## Screenshots 

image

## How this PR could be improved

1. Using another geolocation service to get coordinates since it is IP dependent right now.
2. Displaying the weather of nearby areas as well, to give an idea of the overall weather of the region.

## Checks 

- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.
- [x] I am making a PR with `marcnjaramillo/prep-project-22.P3.2` and not `MLH-Fellowshipprep-project-22.P3.2`. 

## Contributors 

@ThunderBolt-OS 
@samarthya-jha 

## Issues this closes 

This PR closes [issue #3](https://github.com/MLH-Fellowship/prep-project-22.P3.2/issues/3)

### CODE REVIEW

File: src/geolocation.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Geolocation from "@react-native-community/geolocation";
```
Comment (arteevraina) [inline src/geolocation.js]: Can we put this file in some API folder in order to structure it perfectly ?

Comment (marcnjaramillo) [reply src/geolocation.js]: This would be useful. It's not uncommon to have a separate API folder.

**Review by marcnjaramillo — CHANGES_REQUESTED** [review]
This looks great! One change: move the geolocation file into an `/api` directory as suggested by @arteevraina . This would be a good place to keep any other functions that are defined for only interacting with an API.

File: src/geolocation.js (hunk 1 above)
Comment (samarthya-jha) [reply src/geolocation.js]: I've added the `/api` folder and moved `geolocation.js` inside it.

**Review by marcnjaramillo — APPROVED** [review]
Great! Thanks.

**Pull request merged**